app/src/battery.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <device.h>
+#include <init.h>
+#include <kernel.h>
+#include <drivers/sensor.h>
+#include <bluetooth/services/bas.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+struct device *battery;
+
+static int zmk_battery_update(struct device *battery) {
```

Comment:
Should I make this function available to the rest of the application? I guess that's the reason I have the `battery` device passed in to here.

Reply:
I don't see why we would need to, at least not to begin with. This should be able to be pretty self contained.